### ISSUE TITLE
XHR Poll Error after fresh install

### ISSUE THREAD

**Issue by quintz** [pimatic/pimatic#824]
As described in the title I get an XHR Poll error after my 3rd fresh install.

I installed it via the installer (as described here: https://forum.pimatic.org/topic/909/pimatic-installer) on a 32 GB SD Card running the latest Raspbian version.
Then I waited a WHOLE day letting the sqlite3 build up the database.

Now I came home, eagerly awaiting to see my pimatic server running but again: XHR poll error.

The only thing I changed was the IP. I set it to static. I did this while I made the pimatic stop via

`sudo service pimatic stop`

I then rebooted the Pi and started the pimatic server.

I don't know if this should be right but I only can open the pimatic GUI via Port 60. 
But in the config.json it says port 80 (right at the beginning of the config). 
Is maybe there an error?

Here is the log when I ask about 
```
sudo service pimatic status

â pimatic.service - LSB: pimatic service
   Loaded: loaded (/etc/init.d/pimatic)
   Active: active (running) since Fri 2016-03-04 15:26:34 UTC; 6min ago
  Process: 685 ExecStop=/etc/init.d/pimatic stop (code=exited, status=0/SUCCESS)
  Process: 750 ExecStart=/etc/init.d/pimatic start (code=exited, status=0/SUCCESS)
   CGroup: /system.slice/pimatic.service
           ââ756 /usr/bin/nodejs /usr/local/bin/pimatic.js start
           ââ803 npm
           ââ870 sh -c node-pre-gyp install --fallback-to-build
           ââ871 node-pre-gyp
           ââ881 node-gyp

Mar 04 15:26:34 raspberrypi pimatic[750]: Started.
Mar 04 15:26:34 raspberrypi systemd[1]: Started LSB: pimatic service.
```

As I searched for a solution, I stumbled upon the Issue#393. There you mentiond I should try
```
sudo ps alx | grep pimatic
0     0   756     1  20   0 103688 59380 SyS_ep Ssl  ?          1:19 /usr/bin/nodejs /usr/local/bin/pimatic.js start
0     0   975   959  20   0   4536  2132 wait   S    ?          0:00 cc -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_REENTRANT=1 -DSQLITE_THREADSAFE=1 -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_RTREE -DNDEBUG -I/home/pi/pimatic-app/.node-gyp/0.10.24/src -I/home/pi/pimatic-app/.node-gyp/0.10.24/deps/uv/include -I/home/pi/pimatic-app/.node-gyp/0.10.24/deps/v8/include -I./Release/obj/gen/sqlite-autoconf-3090100 -std=c99 -fPIC -Wall -Wextra -Wno-unused-parameter -pthread -include ../src/gcc-preinclude.h -O2 -fno-strict-aliasing -fno-tree-vrp -fno-omit-frame-pointer -MMD -MF ./Release/.deps/Release/obj.target/sqlite3/gen/sqlite-autoconf-3090100/sqlite3.o.d.raw -c -o Release/obj.target/sqlite3/gen/sqlite-autoconf-3090100/sqlite3.o Release/obj/gen/sqlite-autoconf-3090100/sqlite3.c
0     0   976   975  20   0 127860 120240 -     R    ?          4:31 /usr/lib/gcc/arm-linux-gnueabihf/4.9/cc1 -quiet -I /home/pi/pimatic-app/.node-gyp/0.10.24/src -I /home/pi/pimatic-app/.node-gyp/0.10.24/deps/uv/include -I /home/pi/pimatic-app/.node-gyp/0.10.24/deps/v8/include -I ./Release/obj/gen/sqlite-autoconf-3090100 -imultilib . -imultiarch arm-linux-gnueabihf -MMD Release/obj.target/sqlite3/gen/sqlite-autoconf-3090100/sqlite3.d -MF ./Release/.deps/Release/obj.target/sqlite3/gen/sqlite-autoconf-3090100/sqlite3.o.d.raw -MQ Release/obj.target/sqlite3/gen/sqlite-autoconf-3090100/sqlite3.o -D_REENTRANT -D _LARGEFILE_SOURCE -D _FILE_OFFSET_BITS=64 -D _REENTRANT=1 -D SQLITE_THREADSAFE=1 -D SQLITE_ENABLE_FTS3 -D SQLITE_ENABLE_JSON1 -D SQLITE_ENABLE_RTREE -D NDEBUG -include ../src/gcc-preinclude.h Release/obj/gen/sqlite-autoconf-3090100/sqlite3.c -quiet -dumpbase sqlite3.c -march=armv6 -mfloat-abi=hard -mfpu=vfp -mtls-dialect=gnu -auxbase-strip Release/obj.target/sqlite3/gen/sqlite-autoconf-3090100/sqlite3.o -O2 -Wall -Wextra -Wno-unused-parameter -std=c99 -fPIC -fno-strict-aliasing -fno-tree-vrp -fno-omit-frame-pointer -o /tmp/cc3wP2lj.s
0  1000  1135   623  20   0   4268  2008 pipe_w S+   pts/0      0:00 grep --color=auto pimatic

```

Can you see any error?

I'm at the end of my knowledge :/

**Comment by Icesory** [comment]
for testing, you should start pimatic not as deamon.
first, stop the deamon with "sudo service pimatic stop"
and then you can start it with this
```
sudo node /home/pi/pimatic-app/node_modules/pimatic/pimatic.js
```
Now you can see all error direct in your console.

The other thing is, you must configure your config a little bit before it is working.
In the user part you must define password for the admin.
```
  "users": [
    {
      "username": "admin",
      "password": "",
      "role": "admin"
    }
```

**Comment by quintz** [comment]
thanks for the quick reply, Icesory!

I tried stating it as a non-deamon and it worked immediately! Thanks for that!
But I'm wondering why it doesn't work when I start it as deamon...?

Hmm...I already have set my admin password and stuff..

**Comment by Icesory** [comment]
you can now test to start it manually as deamon with
"sudo node /home/pi/pimatic-app/node_modules/pimatic/pimatic.js start"

When this also works your init.d entry is broken

**Comment by quintz** [comment]
Hm...When I manually start the daemon, I also get the XHR poll error...

**Comment by sweetpi** [comment]
Your browser caches the whole frontend. The XHR poll error means, that it can't connect to the deamen (e.g. the deamon is not running or listening on a different port). To get real feedback if the deamon is running you can:

1. Test in an private / incognito window (so the browser does not display the cached page)
2. Check the process list (e.g. `ps -alx`) and the pimatic log file.

**Comment by quintz** [comment]
Thanks for all the advices! 
What I did: 
-Formated the SD card with erase and write method to make it 100% clean.
I now reinstalled pimatic again, step by step and with any caution I could give.
Everything worked fine, I noticed though, that I had some warnings while installing. It warned me, that the graceful-fs wasn't up to date and could cause problems.
After the installation, I launched the server for the first time not as a daemon so I could see if the sqlite3 build up correctly.
After it finished the build-up, I wanted to change the IP from DHCP to static.
I rebooted and wanted to start 
`sudo pimatic.js`
and then this happened
`19:00:27.514 2016-03-07 Monday
19:01:05.878 [pimatic] Starting pimatic version 0.8.101
19:01:12.751 [pimatic] Startup error: Error: PRAGMA synchronous=NORMAL; - SQLITE_BUSY: database is locked
19:01:12.783 [pimatic] exiting...
`

What could this mean?
